src/lib/serialize-integration-markdown.ts

```diff
@@ -3,27 +3,40 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
+import { Plugin } from 'unified'
 import rehypeSurfaceCodeNewlines from '@hashicorp/platform-code-highlighting/rehype-surface-code-newlines'
 import { paragraphCustomAlerts, typography } from '@hashicorp/remark-plugins'
 import rehypePrism from '@mapbox/rehype-prism'
 import { MDXRemoteSerializeResult } from 'next-mdx-remote'
 import { serialize } from 'next-mdx-remote/serialize'
 import rehypeSanitize, { schema } from 'lib/remark-plugins/rehype-sanitize'
-
-// TODO: export types from `next-mdx-remote` v3
-const SERIALIZE_OPTIONS: Parameters<typeof serialize>[1] = {
-	mdxOptions: {
-		remarkPlugins: [paragraphCustomAlerts, typography],
-		rehypePlugins: [
-			[rehypePrism, { ignoreMissing: true }],
-			rehypeSurfaceCodeNewlines,
-			[rehypeSanitize, schema],
-		],
-	},
-}
+import remarkPluginAnchorLinkData from 'lib/remark-plugins/remark-plugin-anchor-link-data'
+// Types
+import type { AnchorLinkItem } from 'lib/remark-plugins/remark-plugin-anchor-link-data'
 
 export default async function serializeIntegrationMarkdown(
-	markdown: string
+	markdown: string,
+	anchorLinks: AnchorLinkItem[] = []
 ): Promise<MDXRemoteSerializeResult> {
+	// TODO: export types from `next-mdx-remote` v3
+	const SERIALIZE_OPTIONS: Parameters<typeof serialize>[1] = {
```

Comment:
Moved the options into the function body here, as we need to pass `anchorLinks` to the anchor links data plugin in order to get the data we need added to that array.